### ISSUE TITLE
Cake Portable upgrade cleanup

### ISSUE THREAD

**Issue by devlead** [cake-build/cake#1052]
Currently files not in nuspec aren't cleaned up when Cake Portable is upgraded, this could potentially be an issue for external dependencies like Roslyn as they're not refreshed so we can be left with old stale assemblies.

One solution for the Cake Portable package would be to add a cleanup script.

**Comment by devlead** [comment]
If we provide an install / uninstall script we could also utilize the `$env:chocolateyPackageParameters` supplied via `--package-parameters` to i.e. clean addins/tools/etc. on demand.

**Comment by gep13** [comment]
@devlead yes, that is certainly an option.  That would infer that the installing user is aware enough of the Cake releases to know when a clean is required.  Not sure if this is the right course of action or not.  Perhaps better to simply always clean, and provide a package parameter to `not clean` so that the more informed user can take that action when they need to.

**Comment by devlead** [comment]
@gep13 agree on that personally would want it to clean every time.

I'm actually wondering if we should change the folder structure of the Cake.Portable package... :wink:
Currently it's just bunch of assemblies in package "root" folder.

It has the side effect that the addins and tools end up outside cake portable in the parent `chocolatey\lib` dir.

If we had similar structure to what most "repos" have `[folder]\Tools\Cake`

then tools would end up in `Tools` and addins in `Tools\Addins`

Would make safer to "clean all" as everything's contained in the same folder.
Just a random thought.

**Comment by gep13** [comment]
>@devlead said...
It has the side effect that the addins and tools end up outside cake portable in the parent chocolatey\lib dir.

That would only happen if you placed the build.cake file in the lib folder though wouldn't it.  I thought that the tools and addins folders were created relative to the location of the build.cake file.

**Comment by gep13** [comment]
@devlead I don't have a Tools folder created in my chocolatey\lib folder.
